### PR TITLE
Add Material.__deepcopy__ method

### PR DESCRIPTION
**Pull request by paulromano**
I needed the ability to copy Materials in some scripting I'm doing and found this missing. Should be pretty straightforward.

### CODE REVIEW

File: openmc/material.py [hunk 1]
```diff
@@ -115,6 +115,30 @@ def __repr__(self):
...
+            clone = type(self).__new__(type(self))
+            clone._id = self._id
+            clone._name = self._name
+            clone._density = self._density
+            clone._density_units = self._density_units
+            clone._nuclides = deepcopy(self._nuclides, memo)
+            clone._elements = deepcopy(self._elements, memo)
+            clone._sab = deepcopy(self._sab, memo)
+            clone._convert_to_distrib_comps = self._convert_to_distrib_comps
+            clone._distrib_otf_file = self._distrib_otf_file
```
Comment (wbinventor) [inline openmc/material.py]: What is the distribution otf file? I think it is a placeholder for DD, should it also be deep copied here?

Comment (paulromano) [inline openmc/material.py]: I don't actually know; it's defined in __init__ which is why I included it here.

Comment (wbinventor) [inline openmc/material.py]: Ok, I think it is safe to also deepcopy it here as a stub for DD.

**Comment by paulromano** [comment]
@wbinventor Did you want me to make any changes?

**Comment by wbinventor** [comment]
Nope, looks good to me!

**Pull request merged**